widgetry/src/widgets/plots.rs

```diff
@@ -0,0 +1,227 @@
+use std::collections::HashSet;
+
+use abstutil::prettyprint_usize;
+use geom::{Circle, Distance, Duration, Percent, Polygon, Pt2D, Time, UnitFmt};
+
+use crate::{Color, EventCtx, GeomBatch, TextExt, Toggle, Widget};
+
+pub struct PlotOptions<X: Axis<X>, Y: Axis<Y>> {
+    pub filterable: bool,
+    pub max_x: Option<X>,
+    pub max_y: Option<Y>,
+    pub disabled: HashSet<String>,
+}
+
+impl<X: Axis<X>, Y: Axis<Y>> PlotOptions<X, Y> {
+    pub fn filterable() -> PlotOptions<X, Y> {
+        PlotOptions {
+            filterable: true,
+            max_x: None,
+            max_y: None,
+            disabled: HashSet::new(),
+        }
+    }
+
+    pub fn fixed() -> PlotOptions<X, Y> {
+        PlotOptions {
+            filterable: false,
+            max_x: None,
+            max_y: None,
+            disabled: HashSet::new(),
+        }
+    }
+}
+
+pub trait Axis<T>: 'static + Copy + std::cmp::Ord {
+    // percent is [0.0, 1.0]
```

Comment:
take it or leave it: call it `ratio` instead of percent?

Reply:
I'd prefer consistency, so this'd also mean changing `geom::Percent` (or rolling it back; still not sure if it's useful). I'll hold off on this for now, but thanks for forcing me to look up the difference between these again.